### DISCUSSION TITLE
Failed to look up terraform-module package in private registry

### DISCUSSION THREAD

**Discussion by nsalfos** [renovatebot/renovate#23539] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate is able to search for custom modules in private terraform regsitry
`app.terraform.io/<my-org>/`

### What you tried so far.

tried a GET request with ThunderClient using the bearer token and it works, but when using the renovate's encrypted im getting unauthorised.

a segment of the 200OK response:
```
{
  "id": "<my-org>/datadog-firehose/aws/0.0.33",
  "owner": "",
  "namespace": "<my-org>",
  "name": "datadog-firehose",
  "version": "0.0.33",
  "provider": "aws",
  "provider_logo_url": "/images/providers/aws.png",
  "description": "",
  "source": "https://github.com/<my-org>/terraform-aws-datadog-firehose",
  "tag": "",
  "published_at": "2023-07-21T16:04:52.080199Z",
  "downloads": 43,
```

(i was able to configure other private (helm/docker + artifactory) registries with encrypted username + token):
this is my `hostRules` block for terraform:

```
"hostRules": [
  {
    "matchHost": "app.terraform.io/<my-org>/",
    "encrypted": {
      "token": "<encrypted-token>"
    }
  }
]
```

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GET https://app.terraform.io/api/registry/v1/modules/<my-org>/datadog-firehose/aws = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=29)
DEBUG: Datasource unauthorized
{
  "datasource": "terraform-module"
  "packageName": "app.terraform.io/<my-org>/datadog-firehose/aws"
  "url": "https://app.terraform.io/api/registry/v1/modules/<my-org>/datadog-firehose/aws"
}

DEBUG: Failed to look up terraform-module package app.terraform.io/<my-org>/datadog-firehose/aws
{
  "dependency": "app.terraform.io/<my-org>/datadog-firehose/aws"
  "packageFile": "main.tf"
}

DEBUG: Failed to look up terraform-module package app.terraform.io/<my-org>/datadog-firehose/aws
{
  "dependency": "app.terraform.io/<my-org>/datadog-firehose/aws"
  "packageFile": "main.tf"
}
```

</details>

**Comment by viceice** [comment]
you forget the https on matchHost. it needs to be only a domain or a full url.

**Comment by nsalfos** [comment]
well... something weird happened. 
suddenly it started working, before i add the https or leave only the domain there... so the conf i posted is wokring.

<img width="907" alt="A8B5A2FE-6012-4927-BC8C-80CC4ABE741C" src="https://github.com/renovatebot/renovate/assets/107048281/c0645d2d-f117-4fc0-b14b-d4eff77f617d">

**Reply by viceice** [reply]
nope, the hostrule is wrong and something else happens

**Reply by nsalfos** [reply]
dunno what to say, here's the full renovate.json.
like i mentioned before, the rules to fetch from artifactory were already working like this :/

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "bumpVersion": "patch",
  "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch"
    }
  ],
  "hostRules": [
    {
      "matchHost": "<my-org>.jfrog.io/artifactory/helm/",
      "hostType": "helm",
      "encrypted": {
        "username": "<redacted>",
        "token": "<redacted>"
      }
    },
    {
      "matchHost": "<my-org>-docker.jfrog.io",
      "hostType": "docker",
      "encrypted": {
        "username": "<redacted>",
        "token": "<redacted>"
      }
    },
    {
      "matchHost": "app.terraform.io/<my-org>/",
      "encrypted": {
        "token": "<redacted>"
      }
    }
  ]
}
```
